app/policies/aal3_policy.rb

```diff
@@ -0,0 +1,15 @@
+class AAL3Policy
+  def initialize(user, sp_session, session)
```

Comment:
`sp_session` is inside of `session` (it's the `:sp` key) .... so passing in both is redundant IMO